app/scripts/modules/core/src/pipeline/config/stages/gremlin/GremlinStageConfig.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from 'react';
+
+import { API } from '@spinnaker/core';
+import { IStageConfigProps, StageConfigField } from 'core/pipeline';
+import { Observable } from 'rxjs';
+import Select from 'react-select';
+
+interface IArgOptions {
+  value?: string;
+}
+
+interface IState {
+  isFetchingData: boolean;
+  commands: any[];
+  targets: any[];
+}
+
+export class GremlinStageConfig extends React.Component<IStageConfigProps> {
+  public state: IState = { isFetchingData: false, commands: [], targets: [] };
+
+  public componentDidMount() {
+    this.checkInitialLoad();
+  }
+
+  private checkInitialLoad = () => {
+    const { stage } = this.props;
+
+    if (stage.gremlinApiKey) {
+      this.fetchAPIData();
+    }
+  };
+
+  private fetchCommands = (apiKey: string) => {
+    return Observable.fromPromise(
+      API.one('gremlin/templates/command')
+        .post({
+          apiKey,
+        })
+        .then((results: any[]) => {
+          return results;
+        })
+        .catch(() => {
+          const results: any[] = [];
+          return results;
+        }),
+    );
+  };
+
+  private fetchTargets = (apiKey: string) => {
+    return Observable.fromPromise(
+      API.one('gremlin/templates/target')
+        .post({
+          apiKey,
+        })
+        .then((results: any[]) => {
+          return results;
+        })
+        .catch(() => {
+          const results: any[] = [];
+          return results;
+        }),
+    );
+  };
+
+  private fetchAPIData = () => {
+    const {
+      stage: { gremlinApiKey },
+    } = this.props;
+
+    this.setState({
+      isFetchingData: true,
+    });
+
+    // Get the data from all the necessary sources before rendering
+    Observable.forkJoin(this.fetchCommands(gremlinApiKey), this.fetchTargets(gremlinApiKey)).subscribe(results => {
+      this.setState({
+        commands: results[0],
+        targets: results[1],
+        isFetchingData: false,
+      });
+    });
+  };
+
+  private onChange = (name: string, value: string) => {
+    this.props.updateStageField({ [name]: value });
+  };
+
+  private handleGremlinCommandTemplateIdChange = (option: IArgOptions) => {
+    this.props.updateStageField({ gremlinCommandTemplateId: option.value });
+  };
+
+  private handleGremlinTargetTemplateIdChange = (option: IArgOptions) => {
+    this.props.updateStageField({ gremlinTargetTemplateId: option.value });
+  };
+
+  public render() {
+    const { stage } = this.props;
+    const { isFetchingData, commands, targets } = this.state;
+
+    // Provides access to meta information for summary box
+    const selectedCommandTemplateMeta = stage.gremlinCommandTemplateId
+      ? commands.find(command => command.guid === stage.gremlinCommandTemplateId)
+      : {};
+    const selectedTargetTemplateMeta = stage.gremlinTargetTemplateId
+      ? targets.find(target => target.guid === stage.gremlinTargetTemplateId)
+      : {};
+
+    return (
+      <React.Fragment>
```

Comment:
If you don't explicitly need to use `<React.Fragment>`, you can just do:
```suggestion
      <>
```